src/types/onyx/TransactionViolation.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @module TransactionViolation
+ * @description Transaction Violation
+ */
+
+/**
+ * Names of the various Transaction Violation types
+ */
+type ViolationName =
+    | 'perDayLimit'
+    | 'maxAge'
+    | 'overLimit'
+    | 'overLimitAttendee'
+    | 'overCategoryLimit'
+    | 'receiptRequired'
+    | 'missingCategory'
+    | 'categoryOutOfPolicy'
+    | 'missingTag'
+    | 'tagOutOfPolicy'
+    | 'missingComment'
+    | 'taxRequired'
+    | 'taxOutOfPolicy'
+    | 'billableExpense';
```

Comment:
Let's include all of them from the get go?
```suggestion
    | 'fieldRequired'
    | 'perDayLimit'
    | 'rter'
    | 'maxAge'
    | 'futureDate'
    | 'overLimit'
    | 'overLimitAttendee'
    | 'overCategoryLimit'
    | 'receiptRequired'
    | 'missingCategory'
    | 'categoryOutOfPolicy'
    | 'missingTag'
    | 'someTagLevelsRequired'
    | 'allTagLevelsRequired'
    | 'tagOutOfPolicy'
    | 'missingComment'
    | 'taxRequired'
    | 'taxOutOfPolicy'
    | 'taxRateChanged'
    | 'taxAmountChanged'
    | 'modifiedAmount'
    | 'receiptNotSmartScanned'
    | 'modifiedDate'
    | 'cashExpenseWithNoReceipt'
    | 'invoiceMarkup'
    | 'duplicatedTransaction'
    | 'smartscanFailed'
    | 'conversionSurcharge'
    | 'autoReportedRejectedExpense'
    | 'nonExpensiworksExpense'
    | 'billableExpense'
    | 'customUnitOutOfPolicy'
    | 'overAutoApprovalLimit'
```